### PR TITLE
Code improvements done

### PR DESCRIPTION
**Pull request by MCCshreyas**
1. Used object initializer syntax.
2. Used string interpolation.
3. Removed unused `usings`.
4. Simplified `if` statement.
5. Removed redundant base controller calls. 
6. Used local function instead of the `Action` delegate. 
7. Use Async overload in async methods.

### CODE REVIEW

File: CycloneDX.Core.Tests/ComponentServiceTests.cs [hunk 1]
```diff
@@ -43,7 +42,7 @@ public async Task RecursivelyGetComponents_ReturnsComponents()
             {
                 new NugetPackage { Name = "Package1", Version = "1.0.0" },
                 new NugetPackage { Name = "Package2", Version = "1.0.0" },
-                new NugetPackage { Name = "Package3", Version = "1.0.0" },
+                new NugetPackage { Name = "Package3", Version = "1.0.0" }
```
Comment (coderpatros) [inline CycloneDX.Core.Tests/ComponentServiceTests.cs:45]: There's no value in a change like this. And by having trailing commas it reduces noise if an additional element is added in the future.

File: CycloneDX.Core.Tests/ComponentServiceTests.cs [hunk 2]
```diff
@@ -14,15 +14,14 @@
...
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
-using Xunit;
-using CycloneDX.Models;
 using CycloneDX.Core.Models;
+using CycloneDX.Models;
 using CycloneDX.Services;
 using Moq;
+using Xunit;
```
Comment (coderpatros) [inline CycloneDX.Core.Tests/ComponentServiceTests.cs:25]: Don't submit changes like this. Just noise without any value.

File: CycloneDX.Core.Tests/ComponentServiceTests.cs (hunk 1 above)
Comment (MCCshreyas) [reply CycloneDX.Core.Tests/ComponentServiceTests.cs:45]: Yeah, right now as there is not element, the commo look like redundant, so I removed that!

File: CycloneDX.Core/Services/ComponentService.cs [hunk 3]
```diff
@@ -33,14 +33,14 @@ public ComponentService(INugetService nugetService)
...
+        /// <param name="nugetPackages"></param>
         /// <returns></returns>
-        public async Task<HashSet<Component>> RecursivelyGetComponentsAsync(IEnumerable<NugetPackage> nugetPackges)
+        public async Task<HashSet<Component>> RecursivelyGetComponentsAsync(IEnumerable<NugetPackage> nugetPackages)
         {
             var components = new HashSet<Component>();
 
             // Initialize the queue with the current packages
-            var packages = new Queue<NugetPackage>(nugetPackges);
+            var packages = new Queue<NugetPackage>(nugetPackages);
```
Comment (coderpatros) [inline CycloneDX.Core/Services/ComponentService.cs:44]: This change has value. Please raise a separate PR for this @MCCshreyas.

File: CycloneDX.Core.Tests/ComponentServiceTests.cs (hunk 2 above)
Comment (MCCshreyas) [reply CycloneDX.Core.Tests/ComponentServiceTests.cs:25]: While submitting PR, I forgot to run tests and I don't know how the the tests projects usings was removed, so I restored them back, then ran tests and then committed the changes!
This is just rearranging usings, not much! 😇

File: CycloneDX.Core/Services/ComponentService.cs (hunk 3 above)
Comment (MCCshreyas) [reply CycloneDX.Core/Services/ComponentService.cs:44]: Separate PR?

File: CycloneDX.Core/Services/NugetService.cs [hunk 4]
```diff
@@ -53,28 +54,16 @@ public class NugetService : INugetService
...
-                if (_fileSystem.File.Exists(currentFilename))
-                {
-                    nuspecFilename = currentFilename;
-                    break;
-                }
-            }
-
-            return nuspecFilename;
+            return _packageCachePaths.Select(packageCachePath => _fileSystem.Path.Combine(packageCachePath, lowerName, version)).Select(currentDirectory => _fileSystem.Path.Combine(currentDirectory, lowerName + ".nuspec")).FirstOrDefault(currentFilename => _fileSystem.File.Exists(currentFilename));
         }
```
Comment (coderpatros) [inline CycloneDX.Core/Services/NugetService.cs:67]: IMHO changes like this make the code less readable.

File: CycloneDX.Xml/XmlBomDeserializer.cs [hunk 5]
```diff
@@ -97,11 +95,12 @@ private static Component GetComponent(XmlNode componentXmlNode, XmlNamespaceMana
...
-
-            license.Id = licenseXmlNode["id"]?.InnerText;
-            license.Name = licenseXmlNode["name"]?.InnerText;
-            license.Url = licenseXmlNode["url"]?.InnerText;
+            var license = new License
+            {
+                Id = licenseXmlNode["id"]?.InnerText,
+                Name = licenseXmlNode["name"]?.InnerText,
+                Url = licenseXmlNode["url"]?.InnerText
+            };
```
Comment (coderpatros) [inline CycloneDX.Xml/XmlBomDeserializer.cs:104]: This is a good tidy up.

File: CycloneDX.Xml/XmlBomDeserializer.cs [hunk 6]
```diff
@@ -111,21 +110,23 @@ private static ComponentLicense GetLicense(XmlNode licenseXmlNode)
...
-
-            organizationalContact.Email = organizationalContactXmlNode["email"]?.InnerText;
-            organizationalContact.Name = organizationalContactXmlNode["name"]?.InnerText;
-            organizationalContact.Phone = organizationalContactXmlNode["phone"]?.InnerText;
+            var organizationalContact = new OrganizationalContact
+            {
+                Email = organizationalContactXmlNode["email"]?.InnerText,
+                Name = organizationalContactXmlNode["name"]?.InnerText,
+                Phone = organizationalContactXmlNode["phone"]?.InnerText
+            };
```
Comment (coderpatros) [inline CycloneDX.Xml/XmlBomDeserializer.cs:130]: This is a good tidy up too.

File: CycloneDX.Xml/XmlBomSerializer.cs [hunk 7]
```diff
@@ -95,9 +95,8 @@ public static string Serialize(Bom bom)
                 if (component.Licenses != null && component.Licenses.Count > 0)
                 {
                     var l = new XElement(ns + "licenses");
-                    foreach (var componentLicense in component.Licenses)
+                    foreach (var license in component.Licenses.Select(componentLicense => componentLicense.License))
                     {
-                        var license = componentLicense.License;
                         if (license.Id != null)
```
Comment (coderpatros) [inline CycloneDX.Xml/XmlBomSerializer.cs:100]: Changes like these can make debugging harder.

**Comment by coderpatros** [comment]
@MCCshreyas there is a lot of stuff going on in this PR. I've added some comments, but to be honest, it is almost impossible to provide meaningful feedback on a PR like this.

There is a lot of noise. There are some changes that provide value. I'm going to close this one as invalid. But if you want to raise more PRs that are more confined in scope I would be happy to review and merge changes that have value.

**Pull request closed without merging**